### PR TITLE
Phase 2 staking

### PR DESCRIPTION
**Pull request by jrwbabylonlab**
What's included in this PR:
- Phase-2 compatible staking txs method from the `Staking` class
- (Phase-1) Observable Staking is now extending the base class `Staking` by adding the OP_RETURN in the script 
- Moved around the files to accomodate this class inheritance changes, including test files

What's NOT included in this PR:
- Any improvement/logic changes to the staking flow, except OP_RETURN
- Slashing
- Documentation updates (README)

### CODE REVIEW

File: src/staking/index.ts [hunk 1]
```diff
@@ -1,754 +1,447 @@
...
+        Buffer.from(stakerPkNoCoordHex, "hex"),
+        [Buffer.from(finalityProviderPkNoCoordHex, "hex")],
+        toBuffers(params.covenantNoCoordPks),
+        params.covenantQuorum,
+        timelock,
+        params.unbondingTime
+      );
+    } catch (error: unknown) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.SCRIPT_FAILURE,
```
Comment (gbarkhatov) [inline src/staking/index.ts:89]: Same here and across other files: https://i.imgur.com/UJmNuB5.png
I suggest formatting the codebase

File: tests/staking/createUnbondingtx.test.ts [hunk 2]
```diff
@@ -27,7 +26,7 @@ describe.each(testingNetworks)("Create unbonding transaction", ({
     stakingTx,
     stakingOutputIndex: 0,
     startHeight: dataGenerator.getRandomIntegerBetween(
-      params.activationHeight, params.activationHeight + 1000,
+      700000, 800000,
```
Comment (gbarkhatov) [inline tests/staking/createUnbondingtx.test.ts:29]: is it a good idea to stick with `700000, 800000`? Can we add a comment there or in the first example

**Review by gbarkhatov — APPROVED** [review]
We also need to remove a comma in `.eslintrc.json`: 
https://i.imgur.com/c2t7gT5.png

File: src/staking/transactions.ts [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+ * @returns {PsbtTransactionResult} The partially signed transaction and the fee.
+ * @throws Will throw an error if the amount or fee rate is less than or equal
+ * to 0, if the change address is invalid, or if the public key is invalid.
+ */
+export function stakingTransaction(
+  scripts: {
+    timelockScript: Buffer;
+    unbondingScript: Buffer;
+    slashingScript: Buffer;
+    dataEmbedScript?: Buffer;
```
Comment (KonradStaniec) [inline src/staking/transactions.ts:58]: if I understand correctly this is now the funciton to build phase-2 staking transaction ? If yes, then this ` dataEmbedScript` is not necessary here, as in phase-2 we do not care about anything other than staking output.

**Review by KonradStaniec — APPROVED** [review]
In general looks good, just one non blocking comment/question

File: src/types/params.ts [hunk 4]
```diff
@@ -7,9 +7,13 @@ export interface Params {
   minStakingAmountSat: number;
   maxStakingTimeBlocks: number;
   minStakingTimeBlocks: number;
+  slashing?: {
```
Comment (vitsalis) [inline src/types/params.ts:10]: Why is slashing optional? To accommodate the observable staking extension?

**Pull request merged**